fotobox/download-app/pkg/cos.go

```diff
@@ -0,0 +1,158 @@
+package pkg
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/IBM/ibm-cos-sdk-go/aws"
+	"github.com/IBM/ibm-cos-sdk-go/aws/credentials/ibmiam"
+	"github.com/IBM/ibm-cos-sdk-go/aws/session"
+	"github.com/IBM/ibm-cos-sdk-go/service/s3"
+)
+
+// Configuration for IBM COS
+var (
+	apiKey            = os.Getenv("apikey")
+	serviceInstanceID = os.Getenv("resource_instance_id")
+	region            = os.Getenv("region") // "eu-de"
+	endpointURL       = os.Getenv("endpointURL") // "https://s3.direct.eu-de.cloud-object-storage.appdomain.cloud" 
+	authEndpoint      = "https://iam.cloud.ibm.com/identity/token"
+	bucketName        = aws.String(os.Getenv("bucket"))
+)
+
+const (
+	thumbnailPrefix = "thumbnail-"
+	downloadDir     = "./images"
+	checkInterval   = 30 * time.Second // Frequency of checking for new images
+)
+
+// Initialize the IBM COS client with IAM authentication
+func initCOSClient() *s3.S3 {
+	// Set up IBM COS session with IAM credentials
+	sess, err := session.NewSession(&aws.Config{
+		Region:           aws.String(region), // Set the region for your COS instance
+		Endpoint:         aws.String(endpointURL),
+		S3ForcePathStyle: aws.Bool(true), // IBM COS requires path-style URLs
+		Credentials: ibmiam.NewStaticCredentials(
+			aws.NewConfig(),
+			authEndpoint,
+			apiKey,
```

Comment:
Have you considered to migrate this to trusted profiles at a later point in time?

Reply:
Yes migration to trusted profiles is planned for a later point in time